### ISSUE TITLE
Training stalls

### ISSUE THREAD

**Issue by alokwarey** [DiffEqML/torchdyn#135]
I am noticing that the training just stalls/stops at a certain epoch for some reason. No errors/explanation. Screenshot below. It got stuck at this epoch and has been here for the last 2 hours. I am not sure why? 

![Untitled](https://user-images.githubusercontent.com/69644995/162642418-7a59272d-9da8-41eb-9634-3f15ceffded0.png)

**Comment by Zymrael** [comment]
This should be because of `PyTorch Lightning`, could you raise `min_epochs` to `100` or so?

**Comment by alokwarey** [comment]
Tried that. No change.

**Comment by Zymrael** [comment]
Could you share your model (and ideally the args passed to `NeuralODE` during init i.e. which solver are you using?) or is the setup the same as the tutorial? This could caused by an unlucky combination of adaptive solver + learning rate + integration time.

**Comment by alokwarey** [comment]
class Learner(pl.LightningModule):
    def __init__(self, t_span:torch.Tensor, model:nn.Module):
        super().__init__()
        self.model, self.t_span = model, t_span        
        
    def forward(self, x):  
        return self.model(x)
    
    def training_step(self, batch, batch_idx):
        x, y = batch      
        t_eval, y_hat = self.model(x, self.t_span)
        y_pred = y_hat[:, 0, :]
        loss = nn.MSELoss()(y_pred, y[0]) 
        return {'loss': loss}   
    
    def configure_optimizers(self):
        return torch.optim.Adam(self.model.parameters(), lr=0.01)

    def train_dataloader(self):
        return trainloader

f = nn.Sequential(nn.Linear(5, 64),
                  nn.ReLU(),                
                  nn.Linear(64, 5))
                                                  
t_span = torch.linspace(0, 1, 800) 

**I tried reducing 800 to 400, 200 and eventually 40. No change. Training is still stalling**

model = NeuralODE(f, sensitivity='adjoint', solver='tsit5', interpolator=None, atol=1e-3, rtol=1e-3)

**Comment by laserkelvin** [comment]
I've also encountered a similar issue as @alokwarey; changing to a fixed-step integrator seems to alleviate the issue, but as far as I'm aware I can't change parameters associated with those solvers (i.e. min/max step size, etc.)

On a quasi-related note, is it possible to print/log diagnostics for the stiffness of the problem, perhaps between training steps? Since `torchdyn` has PyTorch Lightning awareness, one could just expose metrics at a higher level for users to use with PL logging functionality. If this is desirable I could take a crack at it?
